### ISSUE TITLE
Request: Remove virtus dependency

### ISSUE THREAD

**Issue by maxjacobson** [dequelabs/axe-core-gems#402]
## User Story

As a rails developer, when I add axe-core-rspec and axe-core-capybara to my rails app, then virtus is not included in my `Gemfile.lock`.

## Acceptance Criteria

- virtus gem is no longer a dependency

## Technical/Implementation Details

Context for this: virtus has been "discontinued" and is unmaintained: https://github.com/solnic/virtus?tab=readme-ov-file#discontinued

I hadn't noticed this until recently, when I tried to update my Rails app to use Ruby 3.3.5. When I do, I start seeing this warning:

```
/Users/mjacobson/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/virtus-2.0.0/lib/virtus.rb:1: warning: /Users/mjacobson/.rbenv/versions/3.3.5/lib/ruby/3.3.0/ostruct.rb was loaded from the standard library, but will no longer be part of the default gems starting from Ruby 3.5.0.
You can add ostruct to your Gemfile or gemspec to silence this warning.
Also please contact the author of virtus-2.0.0 to request adding ostruct into its gemspec.
```

(virtus uses ostruct)

Given that virtus is not likely to receive any more updates, this warning will continue for now. When Ruby 3.5.0 is released (probably on Christmas 2025, based on the [recent release schedule](https://www.ruby-lang.org/en/downloads/releases/)), axe-core-rspec will not be compatible if it's still depending on virtus.

The maintainer of virtus writes in their discontinuation message that they took what they learned from virtus and brought that to their new set of projects, like [dry-struct](https://github.com/dry-rb/dry-struct). Maybe that could be used instead?

## QA Notes/Instructions

🤷 

## Out of Scope

🤷 

## Dependencies

🤷
